contessa/migration.py

```diff
@@ -0,0 +1,120 @@
+        :param url: the database url where the Alembic migration table is present or planned to be created
+        :param schema: the database schema where the Alembic migration table is present or planned to be created
+        """
+        self.versions_migrations = migrations_map
+        self.package_version = package_version
+        self.url = url
+        self.schema = schema
+        self.conn = Connector(self.url)
+
+    def schema_exists(self):
+        """
+        Check if schema with the Alembic migration table exists.
+        :return: Return true if schema with the Alembic migration exists.
+        """
+        result = self.conn.get_records(
+            f"""
+            SELECT EXISTS (
+               SELECT 1
+               FROM   information_schema.schemata
+               WHERE  schema_name = '{self.schema}'
+            );
+            """
+        )
+        return result.first()[0]
+
+    def migrations_table_exists(self):
+        """
+        Check if the Alembic versions table exists.
+        """
+        result = self.conn.get_records(
+            f"""
+            SELECT EXISTS (
+               SELECT 1
+               FROM   information_schema.tables
+               WHERE  table_schema = '{self.schema}'
+               AND    table_name = '{ALEMBIC_TABLE}'
+           );
+            """
+        )
+        return result.first()[0]
+
+    def get_applied_migration(self):
+        """
+        Get the current applied migration in the target schema.
+        """
+        if self.migrations_table_exists() is False:
+            return None
+        version = self.conn.get_records(f"select * from {self.schema}.{ALEMBIC_TABLE}")
+        return version.first()[0]
+
+    def is_on_head(self):
+        """
+        Check if the current applied migration is valid for the Contessa version.
+        """
+        if self.migrations_table_exists() is False:
+            return False
+        current = self.get_applied_migration()
+
+        fallback_package_version = self.get_fallback_version()
+        return self.versions_migrations[fallback_package_version] == current
+
+    def get_fallback_version(self):
+        """
+        Get fallback version in the case for the Contessa package version do not exist migration.
+        The last package version containing the migration is returned.
+        """
+        keys = list(self.versions_migrations.keys())
+        if self.package_version in self.versions_migrations.keys():
+            return self.package_version
+        if pv(self.package_version) < pv(keys[0]):
+            return list(self.versions_migrations.keys())[0]
+        if pv(self.package_version) > pv(keys[-1]):
+            return list(self.versions_migrations.keys())[-1]
+
+        result = keys[0]
+        for k in keys[1:]:
+            if pv(k) <= pv(self.package_version):
+                result = k
+            else:
+                return result
+
+    """
```

Comment:
This is probably docstring of the following method, probably belongs several lines below.